### PR TITLE
Create CI properties, add maven resources plugin to replace env values

### PR DESCRIPTION
**Pull request by drtechie**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.  

# Type of change  

- [ ] Bug fix  
- [ ] New feature  
- [x] Enhancement  
- [ ] Refactoring  
- [ ] Documentation  
- [x] Other (CI and build changes)

# What has changed?

1. common_${env}.properties file gets filtered with mvn-resources-plugin. 
2. All @@ values are replaced with properties. For example @env.DATABASE_URL@ is replaced by DATABASE_URL present in the environment. 
3. The file is then combined with application.properties by mvn-antrun-plugin.
4. The temporary filtered file generated from steps 1, and 2 is deleted.

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- Copied `common_example.properties` to `common_local.properties`. Edited the DB credentials and was able to start Spring.

# Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by drtechie** [comment]
image

Set an environment variable DATABASE_URL.
After packaging, the value is reflected in `application.properties`.

File: src/main/environment/common_ci.properties [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+foetalMonitorAPIKey = @env.FETOSENSE_API_KEY@
+
+## TM SMS template details(SMS type)
+schedule=TC Schedule SMS
+cancel=TC Cancel SMS
+reSchedule=TC Reschedule SMS
+
+snomedCTPageSize=10
+
+nurseWL=7
```
Comment (drtechie) [inline src/main/environment/common_ci.properties:63]: @Anjana2310  do you know what these IDs are?
Do they change as per the environment?
If yes, I will mark it as an env variable.

Comment (Anjana2310) [reply src/main/environment/common_ci.properties:63]: @drtechie These variables do not vary with environment. snomedCTPageSize=10
 refer to the size of the page to be returned and nurseWL and other WL variables are realted to Worklist implementation which is consistent across all environments.

File: pom.xml [hunk 2]
```diff
@@ -32,7 +32,7 @@
 
 		<environment>${ENV_VAR}</environment>
 		<target-properties>target/classes/application.properties</target-properties>
-		<source-properties>src/main/environment/common_${environment}.properties</source-properties>
+		<source-properties>target/classes/common_${environment}.properties</source-properties>
```
Comment (Anjana2310) [inline pom.xml:35]: @drtechie Can you help me understand why this path change is required? Is it for mvn-antrun-plugin to concatenate?

Comment (drtechie) [reply pom.xml:35]: If we set ENV_VAR=ci

1. `common_ci.properties` is taken first by the mvn-resources-plugin and filtered.
2. In the filtering step, all @@ variables are replaced with environment variables.
3. The filter common_ci.properties is output to target/classes/common_ci.properties.
4. `target/classes/common_ci.properties` is then merged with `application.properties`

This was the only way I could find to make mvn-antrun-plugin and mvn-resources-plugin work together.

Comment (Anjana2310) [reply pom.xml:35]: @drtechie Thankyou for the clarification.

**Review by Anjana2310 — APPROVED** [review]
@drtechie Looks good to me

**Pull request merged**